src/test/scala/guessbook/GuessBookSpec.scala

```diff
@@ -0,0 +1,130 @@
+            game.TheSevenDeathsOfEvelynHardcastle, game.TheWordIsMurder))
+      }
+    }
+
+    "filter to closed circle mysteries if the player correctly guesses that the type of suspect pool is closed circle" in {
+      val stream = new java.io.ByteArrayOutputStream()
+      Console.withOut(stream) {
+        assert(game.poseQuestion(game.TheDecagonHouseMurders, SuspectPoolType, SuspectPool.ClosedCircle,
+          Seq(game.DeathAmongTheUndead, game.DeathInTheHouseOfRain, game.Hyouka, game.TheVillageOfEightGraves)) ===
+          Seq(game.DeathAmongTheUndead, game.DeathInTheHouseOfRain))
+      }
+      assert(stream.toString.strip === "Yes, the selected book's type of suspect pool is closed circle")
+    }
+
+    "throw an exception if the attribute is invalid for the guess value type" in {
+      val invalidGuess1 = intercept[Exception](game.poseQuestion(game.Malice, CountryOfOrigin, true, game.allBooks))
+      assert(invalidGuess1.getMessage === "Invalid attribute for a Boolean guess value")
+
+      val invalidGuess2 = intercept[Exception](game.poseQuestion(game.Malice, DetectiveType, Country.Japan, game.allBooks))
+      assert(invalidGuess2.getMessage === "Invalid attribute for a country guess value")
+
+      val invalidGuess3 = intercept[Exception](game.poseQuestion(game.Malice, Honkaku, GenderEnum.Male, game.allBooks))
+      assert(invalidGuess3.getMessage === "Invalid attribute for a gender guess value")
+
+      val invalidGuess4 = intercept[Exception](game.poseQuestion(game.Malice, SuspectPoolType, Detective.Police, game.allBooks))
+      assert(invalidGuess4.getMessage === "Invalid attribute for a detective type guess value")
+
+      val invalidGuess5 = intercept[Exception](game.poseQuestion(game.Malice, AuthorGender, SuspectPool.ClosedCircle, game.allBooks))
+      assert(invalidGuess5.getMessage === "Invalid attribute for a suspect pool guess value")
+    }
+  }
+
+  "guessBookTitle()" should {
+    "filter the list of books and print the correct result" in {
+      val stream = new java.io.ByteArrayOutputStream()
+
+      Console.withOut(stream) {
+        assert(game.guessBookTitle(game.TheDecagonHouseMurders, "the decagon house murders", 1, game.allBooks)
+          === (Seq(game.TheDecagonHouseMurders), true))
+      }
+      assert(stream.toString.strip === "Player 1 wins! The book was The Decagon House Murders.")
+
+      stream.reset()
+      Console.withOut(stream) {
+        assert(game.guessBookTitle(game.TheDecagonHouseMurders, "Death Among the Undead", 2,
+          Seq(game.DeathAmongTheUndead, game.DeathInTheHouseOfRain, game.TheDecagonHouseMurders, game.OneByOne)) ===
+          (Seq(game.DeathInTheHouseOfRain, game.TheDecagonHouseMurders, game.OneByOne), false))
+      }
+      assert(stream.toString.strip === "Player 2's guess is incorrect. The book is not Death Among the Undead.")
+    }
+
+    "throw an exception if an invalid title is input" in {
```

Comment:
why throw an exception here?

Reply:
Since this method isn't the one that accepts and validates user input, but the one that filters based on the guess, the game shouldn't proceed if an invalid value has reached it (similar to the exceptions thrown for invalid guess attribute in poseQuestion)